ext/auto_flush.c

```diff
@@ -44,7 +44,8 @@ ZEND_RESULT_CODE ddtrace_flush_tracer(bool force_on_startup, bool collect_cycles
     }
 
     if (!get_global_DD_APM_TRACING_ENABLED()) {
-        if (!ddtrace_asm_event_emitted() && !ddtrace_trace_source_is_trace_asm_sourced(&trace) && !ddtrace_standalone_limiter_allow()) {
+        bool allow = ddtrace_standalone_limiter_allow();
+        if (!ddtrace_asm_event_emitted() && !ddtrace_trace_source_is_trace_asm_sourced(&trace) && !allow) {
```

Comment:
you mean to simply reorder the condition?
```
if (!ddtrace_standalone_limiter_allow() && !ddtrace_asm_event_emitted() && !ddtrace_trace_source_is_trace_asm_sourced(&trace))
```
the others don't have side-effects.